tests/testthat/test-update-version.R

```diff
@@ -0,0 +1,33 @@
+test_that("update_version() works", {
+  skip_if_not_installed("rlang", "1.0.1")
+  skip_if_not_installed("testthat", "3.1.2")
+
+  local_demo_project(quiet = TRUE)
+  expect_equal(as.character(desc::desc_get_version()), "0.0.0.9000")
+
+  update_version("dev")
+  expect_equal(as.character(desc::desc_get_version()), "0.0.0.9001")
+
+  update_version("pre-minor")
+  expect_equal(as.character(desc::desc_get_version()), "0.0.99.9000")
+
+  update_version("dev")
+  expect_equal(as.character(desc::desc_get_version()), "0.0.99.9001")
+
+  update_version("minor")
+  expect_equal(as.character(desc::desc_get_version()), "0.1.0")
+
+  expect_snapshot_error(update_version("pre-major"))
```

Comment:
do you agree with this error? this assumes one first needs to go to _dev_.

Reply:
Sure, we can always relax later.